### DISCUSSION TITLE
μSQLite Library Module

### DISCUSSION THREAD

**Discussion by spatialdude** [micropython/micropython#13223] [Show and tell]
Hi

[μSQLite](https://github.com/spatialdude/usqlite) has been updated to v0.1.4 which now includes [SQLite Release 3.44.2](https://sqlite.org/releaselog/3_44_2.html) along with compatability fixes for the latest versions of MicroPython.

**Comment by beyonlo** [comment]
Hello, great work on sqlite for MicroPython. Some questions:

1. Need uSQLite to change something to support the last MicroPython version v1.22?
2. For each new MicroPython release, need something be changed on the uSQLite?
3. Will uSQLite follow the sqlite.org new versions, I mean, implementing in the uSQLite new features too?

Thank you to support new amazing sqlite database on the Micropython!

**Comment by spatialdude** [comment]
@beyonlo [μSQLite](https://github.com/spatialdude/usqlite) has been updated to v0.1.5 which includes the latest version SQLite [**3.46.0**](https://sqlite.org/releaselog/3_46_0.html)

There were some capability fixes to bring it up to date with [MicroPython v1.23](https://github.com/micropython/micropython/releases/tag/v1.23.0). 

I try to routinely keep μSQLite up to date with both MicroPython and SQLite releases.

**Comment by spatialdude** [comment]
[μSQLite](https://github.com/spatialdude/usqlite) has been updated to v0.1.6 which [improves support for exceptions](https://github.com/spatialdude/usqlite/pull/30#issue-2352839405), thanks to @jandriea

**Comment by spatialdude** [comment]
[μSQLite](https://github.com/spatialdude/usqlite) has been updated to v0.1.7 which includes the latest version SQLite [**3.46.1**](https://sqlite.org/releaselog/3_46_1.html)

**Reply by shariltumin** [reply]
Hi, has anyone managed to compile uSQLite in MicroPython v1.24.0-preview-xx-xx?

These are what I did:
1. git clone usqlite in the micropython directory
2. edit esp32_common.cmake to include "usqlite/micropython.cmake"
3. export IDF_PATH="$HOME/disk/esp/esp-idf-522"
4. export IDF_TARGET="esp32"
5. source $IDF_PATH/export.sh
6. make V=1 BOARD=ESP32_GENERIC

3 - 6 are actually being executed by a build script.

The first problem I encountered was:

```
...
FAILED: esp-idf/main_esp32/CMakeFiles/__idf_main_esp32.dir/home/sharil/disk/MP/firmwares/20240904/micropython/usqlite/usqlite_row.c.obj
...
...
/home/sharil/disk/MP/firmwares/20240904/micropython/usqlite/usqlite_row.c: In function 'usqlite_row_type_initialize':
/home/sharil/disk/MP/firmwares/20240904/micropython/usqlite/usqlite_row.c:44:29: error: 'MP_QSTR_Row' undeclared (first use in this function); did you mean 'MP_QSTR_pow'?
   44 |     usqlite_row_type.name = MP_QSTR_Row;
      |                             ^~~~~~~~~~~
      |                             MP_QSTR_pow
...
...
```

I would like to know if anyone has been able to get uSQLite to work in MicroPython and how they have done it.

**Reply by Josverl** [reply]
No, I have not tried it myself,

To halve the problem space though;

    Does the same process work with v1.23.0?

**Reply by shariltumin** [reply]
I try with v1.23.0 and v1.22.0, same problems.

I expect that ```MP_QSTR_Row``` to be defined in ```static const mp_rom_map_elem_t usqlite_module_globals_table[]``` like for example as ```MP_QSTR_connect``` in "usqlite_module.c"

```bash
micropython/usqlite$ egrep MP_QSTR_connect *
grep: scripts: Is a directory
usqlite_cursor.c:            case MP_QSTR_connection:
usqlite_module.c:    { MP_ROM_QSTR(MP_QSTR_connect),                 MP_ROM_PTR(&usqlite_connect_obj) },
micropython/usqlite$ egrep MP_QSTR_row *
grep: scripts: Is a directory
usqlite_connection.c:            case MP_QSTR_row_type:
usqlite_connection.c:            case MP_QSTR_row_type:
usqlite_cursor.c:        case MP_QSTR_row:
usqlite_cursor.c:            case MP_QSTR_rowcount:
```

There are a few undeclared ```MP_QSTR_``` errors during the build process.

Thanks to Elvin Slavik (@spatialdude) for the migration work.
I would be very happy if someone could point out my mistakes. It would be nice to have SQLite in MicroPython (ESP32).

**Reply by spatialdude** [reply]
@shariltumin Not sure what the problem may be on your system, but I did check building the latest Mircropython code on both [Linux](https://github.com/spatialdude/usqlite/wiki/unix) and [Windows](https://github.com/spatialdude/usqlite/wiki/windows) and don't see any compile issues and the module works fine in both. It may be worth trying to compile one of these ports to help troubleshoot the problem.

### Linux
![image](https://github.com/user-attachments/assets/1316193e-b068-4d71-a809-be266f2415e2)

### Windows
![image](https://github.com/user-attachments/assets/9098d89d-4e74-414a-bdf2-7fa224060c04)

**Comment by spatialdude** [comment]
[μSQLite](https://github.com/spatialdude/usqlite) has been updated to v0.1.8 which includes the latest version SQLite [**3.47.0**](https://sqlite.org/releaselog/3_47_0.html)
